addons/web/static/tests/_framework/mock_server/mock_model.js

```diff
@@ -260,16 +287,36 @@ const formatFieldValue = (fields, groupByField, val) => {
         switch (aggregateFunction) {
             // The year is added to the format because is needed to correctly compute the
             // domain and the range (startDate and endDate).
+            case "second_number":
+                return date.second;
+            case "minute_number":
+                return date.minute;
             case "hour":
                 return date.toFormat("HH:00 dd MMM yyyy");
+            case "hour_number":
+                return date.hour;
             case "day":
                 return date.toFormat("yyyy-MM-dd");
+            case "day_of_week":
+                return date.weekday;
+            case "day_of_month":
+                return date.day;
+            case "day_of_year":
+                return date.ordinal;
             case "week":
                 return `W${date.toFormat("WW kkkk")}`;
+            case "iso_week_number":
+                return date.weekNumber;
+            case "month_number":
+                return date.month;
             case "quarter":
                 return `Q${date.toFormat("q yyyy")}`;
+            case "quarter_number":
+                return date.quarter;
             case "year":
                 return date.toFormat("yyyy");
+            case "year_number":
+                return date.year;
```

Comment:
there's a lot of duplicated code here, maybe try to factorize it? Looks like second/minute/hour is obviously for datetime only, but all the rest is the same for both date and datetime